datafusion/physical-plan/src/projection.rs

```diff
@@ -291,7 +453,7 @@ impl ExecutionPlan for ProjectionExec {
     }
 
     fn benefits_from_input_partitioning(&self) -> Vec<bool> {
-        let all_simple_exprs = self.expr.iter().all(|proj_expr| {
+        let all_simple_exprs = self.projection.as_ref().iter().all(|proj_expr| {
```

Comment:
You could also potentially add `Projection::iter()` to iterate directly over exprs from a Projection to avoid the `as_ref()` which might make this code more concise